custom_components/heru/sensor.py

```diff
@@ -103,11 +109,122 @@ async def async_setup_entry(
             entry,
         ),
         HeruNumberSensor("Current exhaust fan power", ICON_FAN, 25, 1, client, entry),
+        HeruPowerSensor("Instantaneous power", entry),
+        HeruEnergySensor("Energy consumption", entry),
     ]
 
     async_add_devices(sensors, update_before_add=True)
 
 
+class HeruPowerSensor(HeruEntity, SensorEntity):
+    """Heru Power Sensor class."""
+
+    _attr_native_unit_of_measurement = "W"
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_device_class = SensorDeviceClass.POWER
+
+    def __init__(self, name: str, entry):
+        _LOGGER.debug("HeruPowerSensor.__init__()")
+        super().__init__(entry)
+        id_name = name.replace(" ", "").lower()
+        self._attr_unique_id = ".".join([entry.entry_id, SENSOR, id_name])
+        self._attr_name = name
+        self._attr_icon = ICON_ENERGY
+        self._device_name = entry.data[CONF_DEVICE_NAME]
+        self._device_model = entry.data[CONF_DEVICE_MODEL]
+        self._heater_p_entity = (
+            "sensor." + self._device_name.lower() + "_current_heating_power"
+        )
+        self._supply_p_entity = (
+            "sensor." + self._device_name.lower() + "_current_supply_fan_power"
+        )
+        self._exhaust_p_entity = (
+            "sensor." + self._device_name.lower() + "_current_exhaust_fan_power"
+        )
+        self._state = None
+
+    async def async_update(self):
+        """async_update"""
+
+        if all(
+            entity in self.hass.states._states.keys()
+            for entity in [
+                self._heater_p_entity,
+                self._supply_p_entity,
+                self._exhaust_p_entity,
+            ]
+        ):
+            heater_p = float(self.hass.states.get(self._heater_p_entity).state)
```

Comment:
Would it be better to subscribe to state changes and disable scan_interval for the new sensors? See https://developers.home-assistant.io/docs/integration_listen_events/ 

Since all sensors are polling every 15 seconds, including the power and energy sensor, it might use "old" values.

Reply:
I agree, the power sensor could be event-based. But I am not sure this could applicable to the energy consumption, if the power is constant over a long period of time you will not see any increase in the energy consumption.
I will implement the event-based trigger for the power, what do you think for the energy sensor?

Reply:
I was thinking of a scenario where all sensors for the HERU component would start polling at the same time. Since the modbus sensor most likely will take longer time to get new values because of network time, the power and energy sensor would finish executing before new values are retrieved from the HERU unit, and thereby using values from the previous poll. 

Having the energy sensor event-based might mess up the calculation as the time is no longer constant? My electro skills are not that good by the way :-)

Lets start with the power sensor for now 👍 Looking forward to testing it!

Reply:
I have implemented the event-based update for the power sensor. I kept the energy consumption sensor's data polling. We would have to track the time if we change it and the data of the energy sensor needs to be updated periodically anyway for the energy dashboard to have the right statistics.
Please test and let me know if there is anything to improve! Thanks, have a great weekend.